frontend/src/simulation/simulation.js

```diff
@@ -0,0 +1,247 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+
+import { getCookie }from "../common";
+
+import "./Simulation.css"
+
+/**
+ * Main component for the simulation.
+ *
+ * Handles all logic, displays information, and makes database query/posts
+ */
+
+// hardcoded list of resources for now
+const resources = [
+    "infrastructure",
+    "electricity",
+    "sanitation",
+    "water",
+    "education",
+];
+
+class Budget extends React.Component {
+    // Once MainView is set up, there will be no state, but rather each will be a prop
+    constructor(props){
+        super(props);
+        this.state = {
+            reaction: null,
+            budgetProposal: {}
+        }
+    }
+
+    componentDidMount() {
+        // for a given list of options set each value in budget proposal to 0
+        let proposal = {};
+        resources.forEach((resource) => {
+            proposal[resource] = 0
+        });
+        this.setState({
+            budgetProposal: proposal,
+        });
+    }
+
+    simulateCitizenResponse = async () => {
+        // check that the total doesn't exceed 100%
+        if (this.validateInput()){
+            const url = '/api/budget_response/';
+            const data = {
+                population: this.props.population,
+                // hardcoded fake data for now
+                budget: this.state.budgetProposal,
+            }
+            const response = await fetch(url, {
+                method: 'POST',
+                body: JSON.stringify(data),
+                headers: {
+                    'Content-type': 'application/json',
+                }
+            });
+            const response_json = await response.json();
+            this.setState({
+                reaction: response_json,
+            });
+        } else { // otherwise throw an alert
+            window.alert("Proposal exceeded budget");
+        }
+    }
+
+    handleSliderOnChange = (e, resource) => {
+        const newVal = e.target.value;
+        const newProposal = this.state.budgetProposal;
+        newProposal[resource] = newVal;
+
+        this.setState({
+            budgetProposal: newProposal,
+        })
+    }
+
+    /*
+    *  Return false is the proposal exceeds the budget
+    */
+    validateInput = () => {
+        let sum = 0;
+        Object.keys(this.state.budgetProposal).forEach((resource) =>
+            sum += parseFloat(this.state.budgetProposal[resource]));
+        return sum <= 1
+    }
+
+
+    render() {
+        let result = 0;
+        if (this.state.reaction)
```

Comment:
no braceless control flow, please! (I thought I'd disallowed this in the linter... will check now and make sure this is disallowed)